webapp/src/main/java/org/apache/atlas/web/resources/AdminResource.java

```diff
@@ -679,6 +681,105 @@ public AtlasImportResult importData(@DefaultValue("{}") @FormDataParam("request"
         return result;
     }
 
+    @POST
+    @Path("/asyncImport")
+    @Produces(Servlets.JSON_MEDIA_TYPE)
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public AtlasAsyncImportRequest importAsync(@DefaultValue("{}") @FormDataParam("request") String jsonData,
+                                               @FormDataParam("data") InputStream inputStream) throws AtlasBaseException {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("==> AdminResource.importAsync(jsonData={}, inputStream={})", jsonData, (inputStream != null));
+        }
+        AtlasAuthorizationUtils.verifyAccess(new AtlasAdminAccessRequest(AtlasPrivilege.ADMIN_IMPORT), "asyncImportData");
+        AtlasAsyncImportRequest asyncImportRequest = null;
+        boolean preventMultipleRequests = true;
+        try {
+            AtlasImportRequest request = AtlasType.fromJson(jsonData, AtlasImportRequest.class);
+            preventMultipleRequests = request != null && request.getOptions() != null
+                    && !request.getOptions().containsKey(AtlasImportRequest.OPTION_KEY_REPLICATED_FROM);
+
+            if (preventMultipleRequests) {
+                acquireExportImportLock("import");
+            }
+            asyncImportRequest = importService.run(request, inputStream, Servlets.getUserName(httpServletRequest),
+                    Servlets.getHostName(httpServletRequest), AtlasAuthorizationUtils.getRequestIpAddress(httpServletRequest));
+        } catch (AtlasBaseException excp) {
+            if (excp.getAtlasErrorCode().equals(AtlasErrorCode.IMPORT_ATTEMPTING_EMPTY_ZIP)) {
+                LOG.info(excp.getMessage());
+                return new AtlasAsyncImportRequest();
+            } else {
+                LOG.error("importData(binary) failed", excp);
```

Comment:
`importData()` => `importAsync()`.

Please review and update other messages as well.